frontstage_api/controllers/secure_messaging_controllers.py

```diff
@@ -1,53 +1,144 @@
+from frontstage_api.exceptions.exceptions import ApiError
 
 
 logger = wrap_logger(logging.getLogger(__name__))
 
 
-def get_messages_list(encoded_jwt):
-    logger.debug('Attempting to retrieve the messages list')
-    method = 'GET'
-    label = request.args.get('label')
+@api_error_handler
+def get_messages_list(encoded_jwt, label):
+    logger.debug('Attempting to retrieve the messages list', label=label)
     url = '{}&label={}'.format(app.config['MESSAGES_LIST_URL'], label)
     headers = {"Authorization": encoded_jwt}
-    response = request_handler(method, url, headers)
+    response = request_handler('GET', url, headers=headers, error_code='FA001')
 
-    # If the response status code is not 200 call exception to stop request
     if response.status_code != 200:
-        logger.error('Error retrieving the messages list', status_code=response.status_code)
-        raise UnexpectedStatusCode(method, url, response.status_code, response.content)
+        logger.error('Error retrieving the messages list', label=label, status_code=response.status_code)
+        raise ApiError('FA001')
 
-    logger.debug('Successfully retrieved the messages list')
-    return json.loads(response.text)
+    logger.debug('Successfully retrieved the messages list', label=label)
+    messages_json = {"messages": json.loads(response.text)}
+    return messages_json
 
 
+@api_error_handler
 def get_unread_message_total(encoded_jwt):
     logger.debug('Attempting to retrieve the unread message total')
-    method = 'GET'
     url = app.config['UNREAD_MESSAGES_TOTAL_URL']
     headers = {"Authorization": encoded_jwt}
-    response = request_handler(method, url, headers, fail=False)
+    response = request_handler('GET', url, headers=headers, error_code='FA002')
 
-    # If request handler has failed to generate a response return the error message
-    if not isinstance(response, Response):
-        return response
-
-    # If the response status code is not 200 return json error message
     if response.status_code != 200:
-        logger.error('Error retrieving the unread messages total', status_code=response.status_code)
-        return {
-            "error": "Unexpected status code returned",
-            "url": url
-        }
+        raise ApiError('FA002')
 
     logger.debug('Successfully retrieved the unread message total')
     return json.loads(response.text)
+
+
+@api_error_handler
+def get_message(encoded_jwt, message_id, label):
+    logger.debug('Attempting to retrieve message', message_id=message_id, label=label)
+    url = app.config['DRAFT_URL'] if label == 'DRAFT' else app.config['MESSAGE_URL']
+    url = '{}/{}'.format(url, message_id)
+    headers = {"Authorization": encoded_jwt}
+    response = request_handler('GET', url, headers=headers, error_code='FA003')
+
+    if response.status_code != 200:
+        logger.error('Error retrieving the messages', status_code=response.status_code, message_id=message_id, label=label)
```

Comment:
logging response text might be helpful. Also I don't think this should be logged at error level as SM can also send 404 response and 403 response here.
Maybe the 404 and 403 responses should be handled differently esp if there is a requirement to display different error messages on the ui?

Reply:
I'd argue a 404 or 403 coming at this point should be an error. 
For 404 any message id that is requested from frontstage will have been generated from the `/messages_list` endpoint. So if we're asking for a msg_id that isn't found something has clearly gone wrong.
Similiarly for the 403 if we have a valid token for the frontstage something must have gone wrong for it to be failing when it reaches secure messaging